src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/CreateIncentive/CreateIncentiveCommandHandler.cs

```diff
@@ -7,28 +7,39 @@
 using System;
 using System.Threading;
 using System.Threading.Tasks;
-using SFA.DAS.EmployerIncentives.Enums;
+using Microsoft.EntityFrameworkCore.Internal;
+using SFA.DAS.EmployerIncentives.Commands.Types.IncentiveApplications;
 
 namespace SFA.DAS.EmployerIncentives.Commands.ApprenticeshipIncentive.CreateIncentive
 {
     public class CreateIncentiveCommandHandler : ICommandHandler<CreateIncentiveCommand>
     {
         private readonly IApprenticeshipIncentiveFactory _apprenticeshipIncentiveFactory;
         private readonly IApprenticeshipIncentiveDomainRepository _apprenticeshipIncentiveDomainRepository;
+        private readonly ICommandPublisher _commandPublisher;
 
         public CreateIncentiveCommandHandler(
             IApprenticeshipIncentiveFactory apprenticeshipIncentiveFactory,
-            IApprenticeshipIncentiveDomainRepository apprenticeshipIncentiveDomainRepository)
+            IApprenticeshipIncentiveDomainRepository apprenticeshipIncentiveDomainRepository,
+            ICommandPublisher commandPublisher)
         {
             _apprenticeshipIncentiveFactory = apprenticeshipIncentiveFactory;
             _apprenticeshipIncentiveDomainRepository = apprenticeshipIncentiveDomainRepository;
+            _commandPublisher = commandPublisher;
         }
 
         public async Task Handle(CreateIncentiveCommand command, CancellationToken cancellationToken = default)
         {
             var existing = await _apprenticeshipIncentiveDomainRepository.FindByApprenticeshipId(command.IncentiveApplicationApprenticeshipId);
             if (existing != null)
             {
+                if (existing.PendingPayments.Count > 0)
```

Comment:
Could these be abstracted into a HasPendingPayment method on the existing object?